tempest/api/workloadmgr/snapshot/test_snapshots.py

```diff
@@ -215,10 +215,110 @@ def test_3_list_snapshot(self):
             reporting.set_test_script_status(tvaultconf.FAIL)
             reporting.test_case_to_write()
 
+    @decorators.idempotent_id('9fe07175-912e-49a5-a629-5f52eeada4c9')
+    @decorators.attr(type='workloadmgr_cli')
+    def test_4_cancel_snapshot(self):
+        try:
+            reporting.add_test_script(str(__name__) + "_cancel_snapshot_cli_with_invalid_options")
+
+            global workload_id
+            LOG.debug("workload is:" + str(workload_id))
+
+            # Create snapshot with CLI command
+            create_snapshot = command_argument_string.snapshot_create + workload_id
+            LOG.debug("Create snapshot command: " + str(create_snapshot))
+            rc = cli_parser.cli_returncode(create_snapshot)
+            if rc != 0:
+                reporting.add_test_step(
+                    "Execute workload-snapshot command with --full",
+                    tvaultconf.FAIL)
+                raise Exception("Command did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload-snapshot command with --full",
+                    tvaultconf.PASS)
+                LOG.debug("Command executed correctly")
+
+            time.sleep(5)
+            snapshot_id = query_data.get_inprogress_snapshot_id(workload_id)
+            LOG.debug("Snapshot ID: " + str(snapshot_id))
+
+            # Snapshot cancel CLI with invalid options
+            snapshot_cancel_novalue = command_argument_string.snapshot_cancel
+            err_msg1 = "workloadmgr snapshot-cancel: error: the following arguments are required: <snapshot_id>"
+            error1 = cli_parser.cli_error(snapshot_cancel_novalue)
+            if error1 and (str(error1.strip('\n')).find(err_msg1) != -1):
+                LOG.debug("Snapshot cancel cli with no snapshot id returned correct error " + str(error1))
+                reporting.add_test_step("Snapshot cancel cli with no option returned correct error",
+                                        tvaultconf.PASS)
+            else:
+                LOG.debug("Snapshot cancel cli with no snapshot id returned no error")
+                reporting.add_test_step("Snapshot cancel cli with no snapshot id returned correct error",
+                                        tvaultconf.FAIL)
+
+            # Snapshot cancel CLI with invalid options
+            snapshot_cancel_invalid = command_argument_string.snapshot_cancel + "invalid"
+            err_msg2 = "ERROR:workloadmgr:No snapshot with a name or ID of 'invalid' exists."
```

Comment:
Can we move this error_msg string to tvaultconf.py?

Reply:
done